### PR TITLE
Include all (beautified) stack trace lines in TAP reporter output

### LINKED ISSUE #2118: Tap reporter not returning valuable info
While integrating with puppetter the stack trace is cool:
```
  ✖ suite › browser › signin › signin.ad › [monitoring, xx] Active Directory User - Rejected with valid user and invalid password Rejected promise returned by test

  1 test failed

  suite › browser › signin › signin.ad › [monitoring, xx] Active Directory User - Rejected with valid user and invalid password

  /Users/giovanni/onelogin/platform/e2e-tests/node_modules/puppeteer/lib/helper.js:231

  Rejected promise returned by test. Reason:

  Error {
    message: 'No node found for selector: button[type=submitss]',
  }

  assert (node_modules/puppeteer/lib/helper.js:231:11)
  DOMWorld.click (node_modules/puppeteer/lib/DOMWorld.js:366:5)
  Frame.<anonymous> (node_modules/puppeteer/lib/helper.js:110:27)
  Page.click (node_modules/puppeteer/lib/Page.js:988:29)
  click (test/suite/browser/signin/signin.ad.spec.js:42:16)

make: *** [test-xx] Error 1
```

But if we use the `tap` formatter, it returns the first element that is not so valuable:
` 'new WaitTask (node_modules/puppeteer/lib/DOMWorld.js:561:28)'` extracted from:
```
TAP version 13
# suite › browser › signin › signin.ad › [monitoring, xx] Active Directory User - Rejected with valid user and invalid password
not ok 1 - suite › browser › signin › signin.ad › [monitoring, xx] Active Directory User - Rejected with valid user and invalid password
  ---
    name: AssertionError
    message: Rejected promise returned by test
    values:
      'Rejected promise returned by test. Reason:': |-
        TimeoutError {
          message: 'waiting for selector "[data-testid="auth-username-screen"]" failed: timeout 30000ms exceeded',
        }
    at: 'new WaitTask (node_modules/puppeteer/lib/DOMWorld.js:561:28)'
  ...


1..1
# tests 1
# pass 0
# fail 1
```

instead `  click (test/suite/browser/signin/signin.ad.spec.js:42:16)`

### PR DESCRIPTION
**Pull request by yovasx2**
Fix #2118 I think it could be better to include full stack trace

### CODE REVIEW

File: lib/reporters/tap.js [hunk 1]
```diff
@@ -41,7 +41,8 @@ function dumpError(error) {
 	}
 
 	if (error.stack) {
-		obj.at = error.stack.split('\n')[0];
+		const errors = error.stack.split('\n');
+		obj.at = errors[errors.length - 1];
```
Comment (tymfear) [inline lib/reporters/tap.js]: You could use 
```javascript
obj.at = errors.pop()
```

Comment (yovasx2) [reply lib/reporters/tap.js]: sure

**Comment by novemberborn** [comment]
> Fix #2118 I think it could be better to include full stack trace

What do you mean by full stack trace? This now takes the last line rather than the first.

**Comment by yovasx2** [comment]
Using puppetter. It returns errors related that framework instead meaningful lines regarding my code

**Comment by yovasx2** [comment]
@novemberborn any update?

**Comment by novemberborn** [comment]
> @novemberborn any update?

I have limited time so it sometimes takes a couple of days before I can reply.

> Using puppeteer. It returns errors related that framework instead meaningful lines regarding my code

Yes but why are you selecting the last line, rather than the first? Maybe it works for your use case but does it break other scenarios?

**Comment by yovasx2** [comment]
I will change it to include all stack: Using it with puppeteer is just giving me a framework's line instead of a meaningful line of my code.

**Comment by yovasx2** [comment]
CC. @sindresorhus

**Comment by yovasx2** [comment]
@novemberborn

**Comment by novemberborn** [comment]
@yovasx2 thanks for your explanation here https://github.com/avajs/ava/issues/2118#issuecomment-499651602

This looks great.

**Pull request merged**